File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[][]} descriptions
 * @return {TreeNode}
 */
var createBinaryTree = function(descriptions) {
    const map = {}
    const set = new Set()
    for (const [parent, child, isLeft] of descriptions) {
        map[parent] = map[parent] || new TreeNode(parent)
        map[child] = map[child] || new TreeNode(child)
        if (isLeft === 1) {
            map[parent].left = map[child]
        } else {
            map[parent].right = map[child]
        }
        set.add("" + child)
    }
    for (const key in map) {
        if (!set.has(key)) {
            return map[key]
        }
    }
};

File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        mp, children = {}, set()
        for parent, child, is_left in descriptions:
            if parent not in mp:
                mp[parent] = TreeNode(parent)
            if child not in mp:
                mp[child] = TreeNode(child)
            if is_left == 1:
                mp[parent].left = mp[child]
            else:
                mp[parent].right = mp[child]
            children.add(child)
        for key in mp:
            if key not in children:
                return mp[key]